torch/_refs/__init__.py

```diff
@@ -5114,6 +5114,19 @@ def bucketize(
     return start.to(dtype=out_dtype)
 
 
+@register_decomposition(aten.geometric)
+@out_wrapper()
+def geometric(self, p, generator=None):
+    assert generator is None
+    tiny = torch.finfo(self.dtype).tiny
+    return self.copy_(
+        torch.floor(
+            torch.log(torch.clamp(torch.rand_like(self), min=tiny)) / math.log1p(-p)
```

Comment:
Actually, I realize `torch.finfo` won't do for integer dtype. So it's probably better to represent `log(1-U)` as `log1p(-U)` similar to as in exponential.